### DISCUSSION TITLE
Download whole album without change photo file name to share friendly format.

### DISCUSSION THREAD

**Discussion by gusmank7** [photoprism/photoprism#4626] [Q&A] [answered]
Based on the documents for [setting.yml.](https://docs.photoprism.app/getting-started/config-files/settings/#file-downloads) We can modify the name for file download with file, original or share.

But I found one issue that still unsolvable, and it stated in the note:

```
Note that your choice will not affect the file names in ZIP archives when you download complete albums, as they always use share-friendly names. However, we may add settings for this in a future release.
I use Photoprism to culling & share event images folder with friends, and most of the time we will just press the download button on the top bar. This result the file name use share-friendly format.
```

As I use my own file name format like **yymmdd-eventname-0001.jpg** it will be great if the global setting also affect the download complete albums/folder.

I found the configuration for this in the source code in `internal/api/download_album.go` file, 

```

fileName := photoprism.FileName(file.FileRoot, file.FileName)
alias := file.ShareBase(0)
key := strings.ToLower(alias)

if seq := aliases[key]; seq > 0 {
	alias = file.ShareBase(seq)
			}

aliases[key] += 1

```
But i don't know anything about Go. I tried a stupid way to change the **ShareBase** to **OriginalBase** but the build-go command return an error. 

So can anyone help me out?

**Comment by graciousgrey** [comment, marked as answer]
It might work to not give the fs.ZipFile function an alias:

`fs.ZipFile(zipWriter, fileName, "", false)` instead of `fs.ZipFile(zipWriter, fileName, alias, false)`

https://github.com/photoprism/photoprism/blob/076a400868f0c77cba6a601dffc00c3feb3e70e3/internal/api/download_album.go#L95

**Reply by gusmank7** [reply]
Work like charm @graciousgrey 👍🏻👍🏻👍🏻 Thank you so much for the help.

Now,I followed all these steps:
```
make docker-build
docker compose up
make terminal
make dep
make build-js
make build-go
./photoprism start

```

Whenever docker is stoped & restarted, the web Server wont start automatically, i have to do **./photoprism start** again before it work.

Is this normal?

**Reply by lastzero** [reply]
The build environment (Docker image) is not intended to be used in production, so it does not automatically start a server. If you want to build an image to use for a local instance, you can e.g. run `make docker-local` to build a new `photoprism/photoprism:local` image that you can specify in a `compose.yaml` file on the same server: https://github.com/photoprism/photoprism/blob/7f3948f4257ba60bb6ecd878facc6cba032237ae/Makefile#L685-L688

**Reply by lastzero** [reply]
Alternatively, you can specify a different entrypoint and/or command in the `compose.yaml` file to run the binary (however, if you pull a fresh develop image, there will be no binary, so the start will fail).
